src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js

```diff
@@ -0,0 +1,103 @@
+import { LinearProgress, makeStyles, Typography } from '@material-ui/core';
+import firebase from 'firebase/app';
+import 'firebase/firestore';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router';
+import ForumBreadcrumbs from './ForumApprovedPostBreadcrumbs';
+import ForumPostDetailsList from './ForumPostDetailsList';
+
+const useStyles = makeStyles((theme) => ({
+  navigationContainer: {
+    marginBottom: theme.spacing(4),
+  },
+}));
+
+export default function ForumApprovedPostDetails() {
+  const classes = useStyles();
+  const { postID } = useParams();
+
+  const [user, setUser] = useState({});
+  const [userLoading, setUserLoading] = useState(false);
+  const [userError, setUserError] = useState(null);
+
+  const [category, setCategory] = useState({});
+  const [categoryLoading, setCategoryLoading] = useState(false);
+  const [categoryError, setCategoryError] = useState(null);
+
+  const [post, setPost] = useState({});
+  const [postLoading, setPostLoading] = useState(false);
+  const [postError, setPostError] = useState(null);
+
+  useEffect(() => {
+    setPostLoading(true);
+    firebase.firestore().collection('forum_posts').doc(postID).get()
+      .then(async (snapshot) => {
+        setPost(snapshot.data());
+      })
+      .catch((err) => {
+        setPostError(err);
+      })
+      .finally(() => {
+        setPostLoading(false);
+      });
+  }, [postID]);
+
+  useEffect(() => {
+    if (post && post.userID) {
+      setUserLoading(true);
+      firebase.firestore().collection('users').doc(post.userID).get()
+        .then((snapshot) => {
+          setUser(snapshot.data());
+        })
+        .catch((err) => {
+          setUserError(err);
+        })
+        .finally(() => {
+          setUserLoading(false);
+        });
+    }
+  }, [post]);
```

Comment:
About the 3 repeated database fetches, I can see that the data necessary for the approved post details in the users (display name) and category (category) collections are already available in approved posts list. Maybe we should be passing this data from the approved posts list to the approved post details, but I'm not sure if that's possible from the way you set up routing.

Reply:
Actually yeah, react-router lets you pass data between screens, totally forgot you could do that!

The only caveat is this data lives in memory, so if you visit this page from a cold start you won't have the data. I'll look into this further and let you know